server/events/vcs/github_client.go

```diff
@@ -509,19 +509,27 @@ func (g *GithubClient) UpdateChecksStatus(ctx context.Context, request types.Upd
 	}
 
 	status, conclusion := g.resolveChecksStatus(request.State)
-	checkRunOutput := github.CheckRunOutput{
-		Title: &request.StatusName,
-	}
-	if request.Output != "" {
-		checkRunOutput.Text = &output
-	}
 
 	// Update checkrun if it exists and if it's not a rerun
 	// request.state is pending only when an operation starts. So, if the checkrun exists and the state is pending, it is a rerun.
 	if checkRun := g.findCheckRun(request.StatusName, checkRuns); checkRun != nil && request.State != models.PendingCommitStatus {
+
+		// Copy the checkrun output if not passed in.
+		var checkRunOutput github.CheckRunOutput
+		if output != "" {
+			checkRunOutput = github.CheckRunOutput{
+				Title: &request.StatusName,
+				Text:  &output,
+			}
+		} else {
+			checkRunOutput = *checkRun.Output
+		}
+
 		summary := request.Description
-		// Append job URL if project command
-		if strings.Contains(request.StatusName, ":") {
+
+		// Append job URL if it is a project plan/apply command
```

Comment:
can u add more details as to why we're doing this for future readers.

Reply:
comments that simply describe the code 1:1 are not super useful imo. they should be used to add context.

Reply:
Thanks for the feedback. Didn't realize my comments were just explaining the what's instead of the why's 😅 Will keep this in mind. I have refactored this and added descriptive comments now